File: my_py_service/my_py_service/pow_two_ints_server.py
from my_interfaces.srv import PowTwoInts

import rclpy
from rclpy.node import Node


class PowTwoIntsServer(Node):

    def __init__(self):
        super().__init__("pow_two_ints_server")
        self.srv = self.create_service(PowTwoInts, "pow_two_ints", self.pow_two_ints_callback)

    def pow_two_ints_callback(self, request, response):
        response.c = 2 ** request.a + 3 ** request.b
        self.get_logger().info('Incoming request\na: %d b: %d' % (request.a, request.b))

        return response


def main(args=None):
    rclpy.init(args=args)

    pow_two_ints_server = PowTwoIntsServer()

    rclpy.spin(pow_two_ints_server)

    rclpy.shutdown()


if __name__ == '__main__':
    main()

File: my_cpp_service/src/pow_two_ints_server.cpp
#include "rclcpp/rclcpp.hpp"
#include "my_interfaces/srv/pow_two_ints.hpp"

#include <memory>
#include <cmath>

void my_pow(const std::shared_ptr<my_interfaces::srv::PowTwoInts::Request> request,
          std::shared_ptr<my_interfaces::srv::PowTwoInts::Response> response)
{
  response->c = std::pow(2, request->a) + std::pow(3, request->b);
  RCLCPP_INFO(rclcpp::get_logger("rclcpp"), "Incoming request\na: %ld" " b: %ld",
                request->a, request->b);
  RCLCPP_INFO(rclcpp::get_logger("rclcpp"), "sending back response: [%ld]", (long int)response->c);
}

int main(int argc, char **argv)
{
  rclcpp::init(argc, argv);

  std::shared_ptr<rclcpp::Node> node = rclcpp::Node::make_shared("pow_two_ints_server");

  rclcpp::Service<my_interfaces::srv::PowTwoInts>::SharedPtr service =
    node->create_service<my_interfaces::srv::PowTwoInts>("pow_two_ints", &my_pow);

  RCLCPP_INFO(rclcpp::get_logger("rclcpp"), "Ready to pow two ints.");

  rclcpp::spin(node);
  rclcpp::shutdown();
}